### PR TITLE
Demo ConfD application (actions) in Zig.

### PR DESCRIPTION
**Pull request by micnovak**

### CODE REVIEW

File: zig-demo-app/app.zig [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+    c_imp.confd_action_set_fd(uinfo, workersock);
+    std.log.info("<== initAction rv={}", .{rv});
+    return rv;
+}
+
+pub fn traceConfdKp(txt: []const u8, kp: [*c]c_imp.confd_hkeypath_t) void {
+    if (kp != null) {
+        var buf: [1024]u8 = undefined;
+        _ = c_imp.confd_pp_kpath(@ptrCast([*c]u8, &buf), buf.len, kp);
```
Comment (jojohans) [inline zig-demo-app/app.zig:44]: I don't know anything about zig, what does this construct mean? Explicitly ignore return value?

File: zig-demo-app/README-zig.adoc [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+----
+var debuglevel: c_uint = c_imp.CONFD_TRACE;
+c_imp.confd_init(EXAMPLE_APP_NAME, c_imp.stderr, debuglevel);
+----
+
+=== C code translation
+
+C code can be translated as starting sample (`goto` and labels should not be used in C code!).
+Translated code is good start, but it has many unnecessary constructs (pointer casts, C lib usage, no error handling) and imports that can be cleaned.
+
```
Comment (jojohans) [inline zig-demo-app/README-zig.adoc:45]: What are you trying to say here? That It's possible to auto translated to zig? If so I think you should say "auto translate". Also, I assume you're saying that the translator can't handle gotos and and labels, I think that could be expressed clearer too.

File: zig-demo-app/app.zig (hunk 1 above)
Comment (micnovak) [reply zig-demo-app/app.zig:44]: Yes, without underscore one gets compilation error.

```
./app.zig:44:29: error: expression value is ignored
        c_imp.confd_pp_kpath(@ptrCast([*c]u8, &buf), buf.len, kp);
```

File: zig-demo-app/README-zig.adoc (hunk 2 above)
Comment (micnovak) [reply zig-demo-app/README-zig.adoc:45]: README was updated.

**Review by jojohans — APPROVED** [review]

**Pull request merged**